lib/improver/ensemble_calibration.py

```diff
@@ -0,0 +1,1418 @@
+    1. Estimate optimised EMOS coefficients from training period.
+    2. Apply optimised EMOS coefficients for future dates.
+
+    """
+    def __init__(self, current_forecast,
+                 historic_forecast, truth,
+                 calibration_method, distribution, desired_units,
+                 predictor_of_mean_flag="mean"):
+        """
+        Create an ensemble calibration plugin that, for Nonhomogeneous Gaussian
+        Regression, calculates coefficients based on historical forecasts and
+        applies the coefficients to the current forecast.
+
+        Parameters
+        ----------
+        current_forecast : Iris Cube or CubeList
+            The Cube or CubeList that provides the input forecast for
+            the current cycle.
+        historic_forecast : Iris Cube or CubeList
+            The Cube or CubeList that provides the input historic forecasts for
+            calibration.
+        truth : Iris Cube or CubeList
+            The Cube or CubeList that provides the input truth for calibration
+            with dates matching the historic forecasts.
+        calibration_method : String
+            The calibration method that will be applied.
+            Supported methods are:
+                ensemble model output statistics
+                nonhomogeneous gaussian regression
+            Currently these methods are not supported:
+                logistic regression
+                bayesian model averaging
+        distribution : String
+            The distribution that will be used for calibration. This will be
+            dependent upon the input phenomenon. This has to be supported by
+            the minimisation functions in
+            ContinuousRankedProbabilityScoreMinimisers.
+        predictor_of_mean_flag : String
+            String to specify the input to calculate the calibrated mean.
+            Currently the ensemble mean ("mean") and the ensemble members
+            ("members") are supported as the predictors.
+
+        """
+        self.current_forecast = current_forecast
+        self.historic_forecast = historic_forecast
+        self.truth = truth
+        self.calibration_method = calibration_method
+        self.distribution = distribution
+        self.desired_units = desired_units
+        self.predictor_of_mean_flag = predictor_of_mean_flag
+
+    def __str__(self):
+        result = ('<EnsembleCalibration: ' +
+                  'calibration_method: {}' +
+                  'distribution: {};' +
+                  'desired_units: {};' +
+                  'predictor_of_mean_flag: {};' +
+                  'current_forecast: {}>' +
+                  'historic_forecast: {}>' +
+                  'truth: {}>')
+        return result.format(
+            self.calibration_method, self.distribution, self.desired_units,
+            self.predictor_of_mean_flag, self.current_forecast,
+            self.historic_forecast, self.truth)
+
+    def process(self):
```

Comment:
My preference is for self.current_forecast to be passed in here instead of in the `__init__` method - same goes for the other classes.